src/nbl/video/COpenGLCommandBuffer.cpp

```diff
@@ -0,0 +1,743 @@
+                ctxlocal->flushStateCompute(gl, SOpenGLContextLocalCache::GSB_ALL, ctxid);
+
+                gl->glCompute.pglDispatchCompute(c.groupCountX, c.groupCountY, c.groupCountZ);
+            }
+            break;
+            case ECT_DISPATCH_INDIRECT:
+            {
+                auto& c = cmd.get<ECT_DISPATCH_INDIRECT>();
+                ctxlocal->nextState.dispatchIndirect.buffer = core::smart_refctd_ptr_static_cast<const COpenGLBuffer>(c.buffer);
+                
+                ctxlocal->flushStateCompute(gl, SOpenGLContextLocalCache::GSB_ALL, ctxid);
+
+                gl->glCompute.pglDispatchComputeIndirect(static_cast<GLintptr>(c.offset));
+            }
+            break;
+            case ECT_DISPATCH_BASE:
+            {
+                auto& c = cmd.get<ECT_DISPATCH_BASE>();
+                // no such thing in opengl (easy to emulate tho)
+                // maybe spirv-cross emits some uniforms for this?
+            }
+            break;
+            case ECT_SET_EVENT:
+            {
+                auto& c = cmd.get<ECT_SET_EVENT>();
+                // TODO
+            }
+            break;
+            case ECT_RESET_EVENT:
+            {
+                auto& c = cmd.get<ECT_RESET_EVENT>();
+                // TODO
+            }
+            break;
+            case ECT_WAIT_EVENTS:
+            {
+                auto& c = cmd.get<ECT_WAIT_EVENTS>();
+                // TODO
+            }
+            break;
+            case ECT_PIPELINE_BARRIER:
+            {
+                auto& c = cmd.get<ECT_PIPELINE_BARRIER>();
+                // TODO
+            }
+            break;
+            case ECT_BEGIN_RENDERPASS:
+            {
+                auto& c = cmd.get<ECT_BEGIN_RENDERPASS>();
+                auto framebuf = core::smart_refctd_ptr_static_cast<const COpenGLFramebuffer>(c.renderpassBegin.framebuffer);
+
+                ctxlocal->nextState.framebuffer.hash = framebuf->getHashValue();
+                ctxlocal->nextState.framebuffer.fbo = std::move(framebuf);
+                ctxlocal->flushStateGraphics(gl, SOpenGLContextLocalCache::GSB_FRAMEBUFFER, ctxid);
+
+                GLuint fbo = ctxlocal->currentState.framebuffer.GLname;
+                beginRenderpass_clearAttachments(gl, c.renderpassBegin, fbo);
+            }
+            break;
+            case ECT_NEXT_SUBPASS:
+            {
+                auto& c = cmd.get<ECT_NEXT_SUBPASS>();
+                // TODO (when we support subpasses) some barriers based on subpass dependencies?
+                // not needed now tho, we dont support multiple subpasses yet
+            }
+            break;
+            case ECT_END_RENDERPASS:
+            {
+                auto& c = cmd.get<ECT_END_RENDERPASS>();
+                // no-op
+            }
+            break;
+            case ECT_SET_DEVICE_MASK:
+            {
+                auto& c = cmd.get<ECT_SET_DEVICE_MASK>();
+                // no-op
+            }
```

Comment:
trying to use this should definitely be an error, and there should be a feture cap in the APIConnection queries for this

Reply:
@achalpandeyy